orinoco/helpers.py

```diff
@@ -7,3 +11,17 @@ def initialize(classes: Iterable[Type[Any]]) -> List[Any]:
 
 def raise_not_provided_field(field_name: str) -> NoReturn:
     raise ValueError("Field {} has to be provided".format(field_name))
+
+
+def _extract_type(value: Union[TypeT, Annotated[TypeT, AnnotationNameT]]) -> Tuple[TypeT, Optional[AnnotationNameT]]:
+    if _is_annotated(value):
+        return value.__origin__, value.__metadata__[0]
+    return value, None
+
+
+def _is_annotated(
+    value: Union[TypeT, Annotated[TypeT, AnnotationNameT]]
+) -> TypeGuard[Annotated[TypeT, AnnotationNameT]]:
```

Comment:
I don't think the value typing on these functions is doing anything, for the `Annotated` it's like creating an instance of the special form instead of defining its structure, and mypy just desugars it into `TypeT`. 
Type hints also aren't strictly subtypes of `type` (e.g. unions)

Reply:
Yeah, I noticed that mypy is not picking it up... However I'd leave it it here for "documentation" what is happening there. I'll try to look into more and find if there is a way how to make it work.
I fixed the type for `TypeT`.